### DISCUSSION TITLE
Password Authentication Error for User (Postgres)

### DISCUSSION THREAD

**Discussion by haranobuhardo** [nocodb/nocodb#2138] [General]
**Please enter the following details**

```
Node: **v16.14.0**
OS: **Win11**
Docker: **false**
Database: **pg**
PackageVersion: **0.90.5**
```

I have followed NOCODB installation instruction and it run successfully with my local PostgreDB, but not with on-premise database which password contains special characters such as "!", "#", "?", Numbers, Chars (Lower and upper case).

I'm running it via NPM (version 8.3.1).

Please see the error log as follows:
`error: password authentication failed for user "pcms_idsu"
    at Parser.parseErrorMessage (D:\nocodb\nocodb-seed\arvel_crynyd\node_modules\pg-protocol\dist\parser.js:287:98)
    at Parser.handlePacket (D:\nocodb\nocodb-seed\arvel_crynyd\node_modules\pg-protocol\dist\parser.js:126:29)
    at Parser.parse (D:\nocodb\nocodb-seed\arvel_crynyd\node_modules\pg-protocol\dist\parser.js:39:38)
    at Socket.<anonymous> (D:\nocodb\nocodb-seed\arvel_crynyd\node_modules\pg-protocol\dist\index.js:11:42)
    at Socket.emit (node:events:520:28)
    at Socket.emit (node:domain:475:12)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
  length: 105,
  severity: 'FATAL',
  code: '28P01',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'auth.c',
  line: '329',
  routine: 'auth_failed'
}`

In Nocodb installation docs, it says that 
`Tip 2: If you plan to input some special characters, you may need to change the character set and collation yourself when creating the database. Please check out the examples for [MySQL Docker](https://github.com/nocodb/nocodb/issues/1340#issuecomment-1049481043).`

But I'm not using any Docker right now (although it produced the same error when I ran it through Docker).

I have tried to look for the solution online, but found none, is this a bug or something?

**Comment by o1lab** [comment]
The character in password of your postgres is not related to the MySQL special character set issue.

Please double confirm if you are entering the right password.

**Reply by haranobuhardo** [reply]
I'm sure it is the correct password.

I have tried to change the username, db address, db name, one by one, and it all produced the same error!

` MESSAGE:                                                                                                              ║
║   password authentication failed for user "pcms_idsu" `

I know this is not a problem with NocoDB, more like its connection to PostgreSQL, but does anyone facing the same issue? My strongest guess is on the special character of the password. Example password "x!D#?? (etc.)"

**Comment by o1lab** [comment]
Moving this to discussions.

**Comment by haranobuhardo** [comment]
### Solved

Solution: Encode the password (which has special characters) with this URL Encoder [tool](https://www.urlencoder.org/)

**Reply by haranobuhardo** [reply]
Facing another problem. I think it is due to I only an access as a limited user. There's no schema created PUBLIC, and my user doesn't have any access to create schema. Thus, I think this is my problem with my administrator now. Or anyone has a way to bypass this schema check?

Error log:
`MESSAGE:                                                                                                                ║
║    CREATE SCHEMA IF NOT EXISTS "public"  AUTHORIZATION "pcms_idsu"  - permission denied for database pgdcmfdbqas  `
